src/compton/CMakeLists.txt

```diff
@@ -9,67 +9,70 @@
 cmake_minimum_required(VERSION 3.0.0)
 project( compton CXX )
 
-##---------------------------------------------------------------------------##
-# Requires COMPTON
-##---------------------------------------------------------------------------##
-find_package( COMPTON QUIET )
-set_package_properties( COMPTON PROPERTIES
-   DESCRIPTION "Access multigroup Compton scattering data."
-   TYPE OPTIONAL
-   PURPOSE "Required for bulding the compton component."
-   )
+# This packages requires libcompton.  See draco/config/vendor_libraries.cmake
+# and draco/config/FindCOMPTON.cmake to manage how the build system discovers
+# this TPL.  If not found, only a stub header file will be installed.
+
+configure_file( ${PROJECT_SOURCE_DIR}/config.h.in
+  ${PROJECT_BINARY_DIR}/config.h )
+
+set( headers
+  ${PROJECT_BINARY_DIR}/config.h
+  ${PROJECT_SOURCE_DIR}/Compton.hh )
 
-if( COMPTON_FOUND )
 # ---------------------------------------------------------------------------- #
 # Source files
 # ---------------------------------------------------------------------------- #
 
-file( GLOB sources *.cc )
-file( GLOB headers *.hh )
+if( COMPTON_FOUND )
+
+set( sources ${PROJECT_SOURCE_DIR}/Compton.cc )
 
-# Make the header files available in the IDE.
-if( MSVC_IDE OR ${CMAKE_GENERATOR} MATCHES Xcode )
-   list( APPEND sources ${headers} )
-endif()
+  # Make the header files available in the IDE.
+  if( MSVC_IDE OR ${CMAKE_GENERATOR} MATCHES Xcode )
+    list( APPEND sources ${headers} )
+  endif()
 
-# ---------------------------------------------------------------------------- #
-# Directories to search for include directives
-# ---------------------------------------------------------------------------- #
+  # -------------------------------------------------------------------------- #
+  # Directories to search for include directives
+  # -------------------------------------------------------------------------- #
 
-include_directories(
-   ${PROJECT_SOURCE_DIR}      # component headers
-   ${PROJECT_BINARY_DIR}      # compton/config.h
-   ${PROJECT_SOURCE_DIR}/..   # ds++ and compton header files
-   ${dsxx_BINARY_DIR}         # ds++/config.h
-   ${ode_BINARY_DIR}          # ode/config.h
-   ${COMPTON_INCLUDE_DIR}
+  include_directories(
+    ${Draco_SOURCE_DIR}/src    # compton/Compton.hh ds++/Assert.hh
+    ${Draco_BINARY_DIR}/src    # compton/config.h
```

Comment:
Is there a reason you prefer this syntax to ${PROJECT_BINARY_DIR}, which would let you include the config.h file without the enclosing compton/ folder? Is it just for clarity, so the #includes in Compton.* are clearly spelled out as "compton/<file>", or is there another reason?

Reply:
* The `include_directories` is deprecated in modern CMake.  I will slowly reduce/eliminate these calls.
* I have started using `${Draco_SOURCE_DIR}/src` to reduce the number of `-I<dir>` commands that appear on the compile line. This change also normalizes directory layout between the build directory and the install directory.